packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -287,7 +304,100 @@ class XCDevice {
     return null;
   }
 
-  List<dynamic> _cachedListResults;
+  /// Observe identifiers (UDIDs) of devices as they attach.
+  ///
+  /// Listening to this stream also allows [detachedDeviceIdentifiers]
+  /// to have listeners. This avoid multiple copies of xcdevice running.
+  Stream<String> attachedDeviceIdentifiers() {
+    if (!isInstalled) {
+      _logger.printTrace("Xcode not found. Run 'flutter doctor' for more information.");
+      return null;
+    }
+    return _attachedDeviceIdentifiers.stream;
+  }
+
+  /// Observe identifiers (UDIDs) of devices as they detach.
+  ///
+  /// [attachedDeviceIdentifiers] stream must have a listener to
+  /// receive detached notifications. This avoid multiple copies
+  /// of xcdevice running.
+  Stream<String> detachedDeviceIdentifiers() {
+    if (!isInstalled) {
+      _logger.printTrace("Xcode not found. Run 'flutter doctor' for more information.");
+      return null;
+    }
+    return _detachedDeviceIdentifiers.stream;
+  }
+
+  // Attach: d83d5bc53967baa0ee18626ba87b6254b2ab5418
+  // Detach: d83d5bc53967baa0ee18626ba87b6254b2ab5418
+  final RegExp _observationIdentifierPattern = RegExp(r'^(\w*): (\w*)$');
+
+  Future<void> _startObservingTetheredIOSDevices() async {
+    try {
+      // Run in interactive mode (via script) to convince
+      // xcdevice it has a terminal attached and pipe stdout.
```

Comment:
Learned this trick from https://github.com/flutter/flutter/pull/12079/files#diff-5a9a07600dceb28cbe887a5a64a24b0aR502-R504.